link.go

```diff
@@ -304,69 +304,59 @@ func (l *link) setSettleModes(resp *frames.PerformAttach) error {
 	return nil
 }
 
-// doFlow handles the logical 'flow' event for a link.
```

Comment:
This is mostly merging `doFlow()` back into `mux()`.  If you hide whitespace changes the diff looks much simpler.